components/gantry/oneaxis/oneaxis.go

```diff
@@ -160,130 +68,67 @@ func newOneAxis(ctx context.Context, deps registry.Dependencies, config config.C
 	}
 
+	board, err := board.FromDependencies(deps, conf.Board)
+	if err != nil {
+		return nil, err
+	}
+
 	oAx := &oneAxis{
-		name:            config.Name,
+		name:            cfg.Name,
 		motor:           _motor,
 		logger:          logger,
-		limitSwitchPins: conf.LimitSwitchPins,
 		lengthMm:        conf.LengthMm,
 		mmPerRevolution: conf.MmPerRevolution,
 		rpm:             conf.GantryRPM,
-		axis:            conf.Axis,
+		board:           board,
 	}
 
-	switch len(oAx.limitSwitchPins) {
-	case 1:
-		oAx.limitType = limitOnePin
-		if oAx.mmPerRevolution <= 0 {
-			return nil, errors.New("gantry with one limit switch per axis needs a mm_per_length ratio defined")
-		}
-
-		board, err := board.FromDependencies(deps, conf.Board)
-		if err != nil {
-			return nil, err
-		}
-		oAx.board = board
-
-		PinEnable := *conf.LimitPinEnabled
-		oAx.limitHigh = PinEnable
-
-	case 2:
-		oAx.limitType = limitTwoPin
-		board, err := board.FromDependencies(deps, conf.Board)
-		if err != nil {
-			return nil, err
-		}
-		oAx.board = board
-
-		PinEnable := *conf.LimitPinEnabled
-		oAx.limitHigh = PinEnable
-
-	case 0:
-		oAx.limitType = limitEncoder
-	default:
-		np := len(oAx.limitSwitchPins)
-		return nil, errors.Errorf("invalid gantry type: need 1, 2 or 0 pins per axis, have %v pins", np)
-	}
-
-	if err := oAx.Home(ctx); err != nil {
-		return nil, err
+	oAx.model, err = oAx.createModel(conf.Axis)
+	if err != nil {
+		return oAx, err
 	}
 
 	return oAx, nil
 }
 
-func (g *oneAxis) Home(ctx context.Context) error {
-	ctx, done := g.opMgr.New(ctx)
-	defer done()
+func (g *oneAxis) createModel(axis r3.Vector) (referenceframe.Model, error) {
+	g.mu.Lock()
+	defer g.mu.Unlock()
+	var errs error
+	m := referenceframe.NewSimpleModel("")
 
-	// Mapping one limit switch motor0->limsw0, motor1 ->limsw1, motor 2 -> limsw2
-	// Mapping two limit switch motor0->limSw0,limSw1; motor1->limSw2,limSw3; motor2->limSw4,limSw5
-	switch g.limitType {
-	// An axis with one limit switch will go till it hits the limit switch, encode that position as the
-	// zero position of the one-axis, and adds a second position limit based on the steps per length.
-	case limitOnePin:
-		err := g.homeOneLimSwitch(ctx)
-		if err != nil {
-			return err
-		}
-	// An axis with two limit switches will go till it hits the first limit switch, encode that position as the
-	// zero position of the one-axis, then go till it hits the second limit switch, then encode that position as the
-	// at-length position of the one-axis.
-	case limitTwoPin:
-		err := g.homeTwoLimSwitch(ctx)
-		if err != nil {
-			return err
-		}
-	// An axis with an encoder will encode the
-	case limitEncoder:
-		err := g.homeEncoder(ctx)
-		if err != nil {
-			return err
-		}
-	}
-	return nil
-}
+	f, err := referenceframe.NewStaticFrame(g.name, spatialmath.NewZeroPose())
+	errs = multierr.Combine(errs, err)
+	m.OrdTransforms = append(m.OrdTransforms, f)
```

Comment:
Naive question: if there are errors early on, why continue to accumulate them, rather than returning early?

Reply:
We'd like to get as much information on a reference frame out as possible. So that motion can use the pertinent information it can if it needs.